Sources/IfTag.swift

```diff
@@ -100,21 +107,68 @@ final class IfExpressionParser {
   let tokens: [IfToken]
   var position: Int = 0
 
-  init(components: [String], tokenParser: TokenParser) throws {
-    self.tokens = try components.map { component in
-      if let op = findOperator(name: component) {
-        switch op {
-        case .infix(let name, let bindingPower, let cls):
-          return .infix(name: name, bindingPower: bindingPower, op: cls)
-        case .prefix(let name, let bindingPower, let cls):
-          return .prefix(name: name, bindingPower: bindingPower, op: cls)
+  private init(tokens: [IfToken]) {
+    self.tokens = tokens
+  }
+  
+  static func parser(components: [String], tokenParser: TokenParser) throws -> IfExpressionParser {
+    return try IfExpressionParser(components: ArraySlice(components), tokenParser: tokenParser)
+  }
+
+  private init(components: ArraySlice<String>, tokenParser: TokenParser) throws {
+    var parsedComponents = Set<Int>()
+    var bracketsBalance = 0
+    self.tokens = try zip(components.indices, components).flatMap { (index, component) in
+      guard !parsedComponents.contains(index) else { return nil }
+
+      if component == "(" {
+        bracketsBalance += 1
+        let (expression, parsedCount) = try IfExpressionParser.subExpression(
+          from: components.suffix(from: index + 1),
+          tokenParser: tokenParser
+        )
+        parsedComponents.formUnion(Set(index...(index + parsedCount)))
+        return .subExpression(expression)
+      }
+      else if component == ")" {
+        bracketsBalance -= 1
+        if bracketsBalance < 0 {
+          throw TemplateSyntaxError("'if' expression error: missing opening bracket")
+        }
+        parsedComponents.insert(index)
+        return nil
+      } else {
+        parsedComponents.insert(index)
+        if let op = findOperator(name: component) {
+          switch op {
+          case .infix(let name, let bindingPower, let cls):
+            return .infix(name: name, bindingPower: bindingPower, op: cls)
+          case .prefix(let name, let bindingPower, let cls):
+            return .prefix(name: name, bindingPower: bindingPower, op: cls)
+          }
         }
+        return .variable(try tokenParser.compileFilter(component))
       }
-
-      return .variable(try tokenParser.compileFilter(component))
     }
   }
 
+  private static func subExpression(from components: ArraySlice<String>, tokenParser: TokenParser) throws -> (Expression, Int) {
+    var bracketsBalance = 1
+    let subComponents = components
+      .prefix(while: {
+        if $0 == "(" { bracketsBalance += 1 }
```

Comment:
Not a big fan of having the content of the `if` on the same line, and pretty sure `swiftlint`'ll complain about this too.